atlasdb-config/src/test/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplierTest.java

```diff
@@ -50,23 +51,42 @@
 
     @Test
     public void delegateToFactoriesAnnotatedWithAutoServiceForCreatingKeyValueServices() {
-        ServiceDiscoveringAtlasSupplier atlasSupplier = new ServiceDiscoveringAtlasSupplier(
-                metrics, kvsConfig, leaderConfig);
+        ServiceDiscoveringAtlasSupplier atlasSupplier = new ServiceDiscoveringAtlasSupplier(metrics,
+                kvsConfig,
+                Optional::empty,
+                leaderConfig,
+                Optional.empty(),
+                Optional.empty(),
+                AtlasDbConstants.DEFAULT_INITIALIZE_ASYNC,
+                AtlasDbFactory.THROWING_FRESH_TIMESTAMP_SOURCE);
```

Comment:
In general for these creators it seems

```java
new ServiceDiscoveringAtlasSupplier(
                metrics,
                kvsConfig,
                leaderConfig,
                Optional.empty(),
                Optional.empty())
```

will do, or possibly extracting a static method in here that just takes the 3 original params

Reply:
I think I actually wanted to get rid of that constructor, I can't remember why I didn't though. I'll make a helper method inside the test however.